.github/workflows/fork-connector-tests.yml

```diff
@@ -0,0 +1,138 @@
+
+jobs:
+  check-fork-and-trigger-tests:
+    name: Check Fork and Trigger Tests
+    runs-on: ubuntu-24.04
+    if: github.event.pull_request.head.repo.full_name != github.repository
+    steps:
+      - name: Check if fork is from outside airbytehq org
+        id: check-fork
+        run: |
+          FORK_REPO="${{ github.event.pull_request.head.repo.full_name }}"
+          echo "fork-repo=$FORK_REPO" >> $GITHUB_OUTPUT
+
+          if [[ "$FORK_REPO" == airbytehq/* ]]; then
+            echo "is-external-fork=false" >> $GITHUB_OUTPUT
+            echo "Fork is from airbytehq org, skipping"
+          else
+            echo "is-external-fork=true" >> $GITHUB_OUTPUT
+            echo "Fork is external: $FORK_REPO"
+          fi
+
+      - name: Get fork owner
+        id: fork-owner
+        if: steps.check-fork.outputs.is-external-fork == 'true'
+        run: |
+          FORK_OWNER=$(echo "${{ steps.check-fork.outputs.fork-repo }}" | cut -d'/' -f1)
+          echo "owner=$FORK_OWNER" >> $GITHUB_OUTPUT
+          echo "Fork owner: $FORK_OWNER"
+
+      - name: Attempt to get GitHub App token for fork
+        id: get-app-token
+        if: steps.check-fork.outputs.is-external-fork == 'true'
+        continue-on-error: true
+        uses: actions/create-github-app-token@v2
+        with:
+          app-id: ${{ secrets.OCTAVIA_BOT_APP_ID }}
+          private-key: ${{ secrets.OCTAVIA_BOT_PRIVATE_KEY }}
+          owner: ${{ steps.fork-owner.outputs.owner }}
+
+      - name: Post pending commit status
+        if: steps.get-app-token.outcome == 'success'
+        run: |
+          curl -X POST \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github+json" \
+            "${{ github.api_url }}/repos/${{ github.repository }}/statuses/${{ github.event.pull_request.head.sha }}" \
+            -d '{
+              "state": "pending",
+              "context": "Fork Connector Tests",
+              "description": "Running connector tests in fork with BYO secrets...",
+              "target_url": "${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"
+            }'
+
+      - name: Trigger connector tests in fork
+        id: trigger-tests
+        if: steps.get-app-token.outcome == 'success'
+        uses: the-actions-org/workflow-dispatch@v4
+        with:
+          token: ${{ steps.get-app-token.outputs.token }}
+          repo: ${{ steps.check-fork.outputs.fork-repo }}
+          workflow: connector-ci-checks.yml
+          ref: ${{ github.event.pull_request.head.ref }}
+          wait-for-completion: true
+          wait-for-completion-timeout: 2h
+          wait-for-completion-interval: 30s
+          inputs: |
+            {
+              "repo": "${{ steps.check-fork.outputs.fork-repo }}",
+              "gitref": "${{ github.event.pull_request.head.ref }}",
+              "pr": "${{ github.event.pull_request.number }}"
+            }
```

Comment:
Our token is only good for 60 minutes. Make a comment about this preceeding this step, and reduce timeout to 55 minutes.